bluemira/base/components.py

```diff
@@ -107,21 +108,74 @@ def get_component(
 
         Notes
         -----
-            This function is just a wrapper of the anytree.search.findall_by_attr
+            This function is just a wrapper of the anytree.search.findall
             function.
         """
-        if full_tree:
-            found_components = anytree.search.findall_by_attr(self.root, name)
-        else:
-            found_components = anytree.search.findall_by_attr(self, name)
+        return self._get_thing(
+            lambda n: anytree.search._filter_by_name(n, "name", name), first, full_tree
+        )
 
-        if len(found_components) == 0:
-            return None
+    def get_component_properties(
+        self, properties: Union[Iterable[str], str], first=True, full_tree=False
+    ):
+        """
+        Get properties from a component
+
+        Parameters
+        ----------
+        name: str
+            The name of the component to search for.
+        first: bool
+            If True, only the first element is returned, by default True.
+        full_tree: bool
+            If True, searches the tree from the root, else searches from this node, by
+            default False.
+
+        Returns
+        -------
+        Lists of properties of components
```

Comment:
We could be clearer here that this returns a list if `first`, but a tuple of lists if `not first`. Or am I misunderstanding?

Reply:
Update, hopefully thats clearer